pyclesperanto_prototype/_tier8/_AffineTransform3D.py

```diff
@@ -157,20 +157,22 @@ def shear_z(self,angle_x_in_degrees: float = 0, angle_y_in_degrees: float = 0 ):
         """          
         import math
         try:
-            shear_factor_xz = 1.0 / math.tan(angle_x_in_degrees * math.pi / 180)
+            shear_factor_xz = 1.0 / math.tan((angle_x_in_degrees-90) * math.pi / 180)
         except ZeroDivisionError:
-            shear_factor_xz = 0
+            print("Error1")
+            shear_factor_xz = math.pi
         
         try:
-            shear_factor_yz = 1.0 / math.tan(angle_y_in_degrees * math.pi / 180)
+            shear_factor_yz = 1.0 / math.tan((angle_y_in_degrees-90) * math.pi / 180)
         except ZeroDivisionError:
-            shear_factor_yz = 0
+            print("Error2")
+            shear_factor_yz = math.pi
 
         # shearing
-        self._concatenate(np.asarray([
-            [1, 0, shear_factor_xz, 0],
-            [0, 1, shear_factor_yz, 0],
-            [0, 0 , 1, 0],
+        self._pre_concatenate(np.asarray([
```

Comment:
Why do you use a pre_concatenate for the shear matrices?

Reply:
Because we describe a forward transform (from source image to target), but we apply a backward-transfrom (from every pixel in target, we have to find out where the source pixel is). This is equivalent to a .inverse() operation.

Goal is to define an API that is intuitive from the outside (saying, I want shear this and rotate that) and then internally do the math right so that the result image is correct, according to what the user wrote.

Reply:
Thanks
I think I'm still confused, you perform the inverse of the affine transform here, right?
[affine_transform.py](https://github.com/clEsperanto/pyclesperanto_prototype/blob/58e9f2239a31add046d539c4483e3e9798b0886b/pyclesperanto_prototype/_tier8/_affine_transform.py#L75)?
Apologies if this is getting too messy!